base/sim/FairPrimaryGenerator.cxx

```diff
@@ -153,6 +153,19 @@ Bool_t FairPrimaryGenerator::Init()
     return kTRUE;
 }
 
+
+Bool_t FairPrimaryGenerator::Finish()
+{
+    /** Initialize list of generators*/
+    for (Int_t i = 0; i < fGenList->GetEntries(); i++) {
+        FairGenerator *gen = static_cast<FairGenerator *>(fGenList->At(i));
+        if (gen) {
+            gen->Finish();
+        }
+    }
```

Comment:
Maybe use TRangeDynCast?
I know, it's doing a `dynamic_cast`, which means a bit of performance. But this is in FinishRun, so maybe not as bad?

I think, in the long run, we should switch to `std::vector`.

```suggestion
    for (auto gen : TRangeDynCast<FairGenerator>(fGenList)) {
        if (gen) {
            gen->Finish();
        }   
    }
```